ingestion/src/metadata/ingestion/source/dashboard/tableau/models.py

```diff
@@ -121,6 +128,9 @@ class UpstreamTable(BaseModel):
 class DataSource(BaseModel):
     id: str
     name: Optional[str] = None
+    description: Optional[str] = None
+    hasExtracts: Optional[bool] = None
+    tags: Optional[List[TableauDataModelTag]] = []
```

Comment:
We should assign `None` here as the default and handle the empty list on the metadata.py side

Reply:
I can set None, but I don't fully agree that should be None here.
If you look into [TableauDashboard](https://github.com/open-metadata/OpenMetadata/blob/1924f44b50765e275037868455e5c693544596b3/ingestion/src/metadata/ingestion/source/dashboard/tableau/models.py#L170), we use a list of empty tags.

When dealing with list, I generally prefer to use empty list, vs None objects.

Reply:
Makes sense it should be fine